library/network_connections.py

```diff
@@ -805,6 +807,11 @@ def connection_create(self, connections, idx, connection_current=None):
             )
             s_wired = self.connection_ensure_setting(con, NM.SettingWired)
             s_wired.set_property(NM.SETTING_WIRED_MAC_ADDRESS, connection["mac"])
+            s_con.set_property(
+                NM.SETTING_CONNECTION_TYPE, NM.SETTING_MATCH_SETTING_NAME
+            )
```

Comment:
this is wrong. You are setting here `connection.type=match`, but that is not right. `connection.type` is the main type of the profile, usually something like bridge or ethernet. A `[match]` section is only an optional part of any profile, it's never the main type of the profile.

Reply:
I see, thanks